src/sentry/identity/github/provider.py

```diff
@@ -18,7 +18,14 @@ def get_user_info(access_token):
     return resp
 
 
+# Github has 2 types of apps -- Github apps and OAuth apps. SSO is implemented
+# using OAuth App, but signup and integrations use the github app. When github
+# apps have API parity with OAuth apps, we should move SSO to it as well.
+# https://developer.github.com/apps/differences-between-apps/
+
+
 class GitHubIdentityProvider(OAuth2Provider):
+    # `key` goes into the db so you can't just change it.
```

Comment:
hmm this is technically not always true. e.g. I'm creating a provider to augment the identity provider with the new-account flow, and I need to use a different key so it can be fetched from redis, but this key is never stored in the db (only 'github' is)